File: assignments/2026/week10/PGM_42576_LGJ.java
// PGM - 42576 완주하지못한 선수
// 해쉬
// https://school.programmers.co.kr/learn/courses/30/lessons/42576?language=java


import java.util.HashMap;

class Solution {

    public String solution(String[] participant, String[] completion) {

        HashMap<String, Integer> map = new HashMap<>();

        for (String p : participant)
            map.put(p, map.getOrDefault(p, 0) + 1);

        for (String c : completion)
            map.put(c, map.get(c) - 1);

        for (var entry : map.entrySet()) {
            if (entry.getValue() != 0)
                return entry.getKey();
        }

        return "";
    }
}

File: assignments/2026/week10/PGM_42576_LGJ.py
# PGM - 42576 완주하지못한 선수
# 해쉬
# https://school.programmers.co.kr/learn/courses/30/lessons/42576?language=java

def solution(participant, completion):
	participantCnt = {}

	for p in participant:
		participantCnt[p] = participantCnt.get(p, 0)+1

	for c in completion:
		participantCnt[c] -= 1

	for key, value in participantCnt.items():
		if participantCnt[key] > 0:
			return key

	return ""

